src/transaction/log_applier.c

```diff
@@ -3686,7 +3689,40 @@ la_make_room_for_mvcc_insid (RECDES * recdes)
 
   assert (recdes->area_size >= recdes->length + OR_MVCCID_SIZE);
 
-  LA_MOVE_INSIDE_RECORD (recdes, OR_INT_SIZE + OR_MVCCID_SIZE, OR_INT_SIZE);
+  //LA_MOVE_INSIDE_RECORD (recdes, OR_INT_SIZE + OR_MVCCID_SIZE, OR_INT_SIZE);
+  LA_MOVE_INSIDE_RECORD (recdes, OR_MVCC_INSERT_ID_OFFSET + OR_MVCCID_SIZE, OR_MVCC_INSERT_ID_OFFSET);
```

Comment:
```suggestion
  LA_MOVE_INSIDE_RECORD (recdes, OR_MVCC_INSERT_ID_OFFSET + OR_MVCC_INSERT_ID_SIZE, OR_MVCC_INSERT_ID_OFFSET);
```
1. OR_MVCC_INSERT_ID_SIZE seems more proper.
2. I think your change is correct. The existing code seems to move the record not from insert_mvcc_id, but from chn. I don't really know about `chn`, but it would've acted up. The only concern I have is somewhere else might have some fixes for this (I think it's not :)). Anyway, I agree with this change.
The record header format:
`RepID(3) | flags(1) | chn (4) | insert_mvcc_id | ...`

Reply:
Thanks to check second one.